File: src/main/java/Piruru/cards/ScoutAttacks.java
package Piruru.cards;

import Piruru.actions.ScoutCardsAction;
import Piruru.actions.ScoutFollowUpAction;
import com.megacrit.cardcrawl.actions.utility.WaitAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import static Piruru.Piruru.makeID;

public class ScoutAttacks extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(ScoutAttacks.class.getSimpleName()));
    private static final int COST = 0;
    private static final CardType TYPE = CardType.SKILL;
    private static final CardRarity RARITY = CardRarity.BASIC;
    private static final CardTarget TARGET = CardTarget.SELF;
    private static final int SCOUT_NUM = 3;
    private static final int SCOUT_NUM_UP = 3;

    public ScoutAttacks() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, SCOUT_NUM_UP, COST);
        baseMagicNumber = magicNumber = SCOUT_NUM;
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        act(new ScoutCardsAction(p, magicNumber));
        act(new WaitAction(0.4F));
        act(new ScoutFollowUpAction());
    }
}

File: src/main/kotlin/Cirno/cards/ScoutAttacks.kt
package Cirno.cards

import Cirno.Cirno.Statics.makeID
import Cirno.abstracts.PiruruCard
import Cirno.actions.ScoutCardsAction
import Cirno.actions.ScoutFollowUpAction
import com.megacrit.cardcrawl.actions.utility.WaitAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.monsters.AbstractMonster


class ScoutAttacks : PiruruCard(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, SCOUT_NUM_UP, COST) {
    init {
        magicNumber = SCOUT_NUM
        baseMagicNumber = magicNumber
    }

    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        act(ScoutCardsAction(p, magicNumber))
        act(WaitAction(0.4f))
        act(ScoutFollowUpAction())
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(ScoutAttacks::class.java))
        private val COST = 0
        private val TYPE = AbstractCard.CardType.SKILL
        private val RARITY = AbstractCard.CardRarity.BASIC
        private val TARGET = AbstractCard.CardTarget.SELF
        private val SCOUT_NUM = 3
        private val SCOUT_NUM_UP = 3
    }
}
